index.bs

```diff
@@ -1552,6 +1574,114 @@ To <dfn>parse a filter pair</dfn> given a [=map=] |map|:
+1. Let |unspecifiedContribution| be null.
+1. [=list/iterate|For each=] |data| of |dataList|:
+    1. If |data| is not a [=map=], return null.
+    1. If |data|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"]
+        does not [=map/exist=], return null.
+    1. If |data|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"]
+        is not a [=string=], return null.
+    1. Let |dataKeyPiece| be the result of running [=parse an aggregation key piece=]
+        with |data|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"].
+    1. If |dataKeyPiece| is an error, return null.
+    1. If |data|["<code>[=aggregate-debug-reporting JSON key/value=]</code>"]
+        does not [=map/exist=], return null.
+    1. If |data|["<code>[=aggregate-debug-reporting JSON key/value=]</code>"] is
+        not an integer or is less than or equal to 0 or is greater than |maxValue|, return null.
+    1. Let |contribution| be a new [=aggregatable contribution=] with the items:
+        : [=aggregatable contribution/key=]
+        :: |dataKeyPiece|
+        : [=aggregatable contribution/value=]
+        :: |data|["<code>[=aggregate-debug-reporting JSON key/value=]</code>"]
+    1. If |data|["<code>[=aggregate-debug-reporting JSON key/types=]</code>"] does
+        not [=map/exist=], return null.
+    1. Let |dataTypes| be |data|["<code>[=aggregate-debug-reporting JSON key/types=]</code>"].
+    1. If |dataTypes| is not a [=list=] or [=list/is empty=], return null.
+    1. [=list/iterate|For each=] |type| of |dataTypes|:
+        1. If |type| is not a [=string=], return null.
+        1. If |type| is "`unspecified`":
+            1. If |unspecifiedContribution| is not null, return null.
+            1. Set |unspecifiedContribution| to |contribution|.
+            1. [=iteration/Continue=].
+        1. If |supportedTypes| does not [=set/contain=] |type|, [=iteration/continue=].
+        1. If |debugDataMap|[|type|] [=map/exists=], return null.
+        1. [=map/Set=] |debugDataMap|[|type|] to |contribution|.
+1. If |unspecifiedContribution| is null, return |debugDataMap|.
+1. [=set/iterate|For each=] |type| of |supportedTypes|:
+    1. If |debugDataMap|[|type|] does not [=map/exist=], [=map/set=]
+        |debugDataMap|[|type|] to |unspecifiedContribution|.
+1. Return |debugDataMap|.
+
+To <dfn>parse aggregate debug reporting config</dfn> given a [=map=] |map|, a
+positive integer |maxValue|, a [=boolean=] |isSource|, and an [=aggregate debug reporting config=]
+|default|:
+
+1. If |map|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"] does not
+    [=map/exist=], return |default|.
+1. If |map|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"] is not a
+    [=string=], return |default|.
+1. Let |keyPiece| be the result of running [=parse an aggregation key piece=]
+    with |map|["<code>[=aggregate-debug-reporting JSON key/key_piece=]</code>"].
+1. If |keyPiece| is an error, return |default|.
```

Comment:
Are we intentionally ignoring the errors in this algorithm? If so, could we add a non-normative note explaining why?

Reply:
Added note.